### DISCUSSION TITLE
Vercel Edge Runtime (Middleware) Verification using SPKI

### DISCUSSION THREAD

**Discussion by christianmittler** [panva/jose#677] [Q&A]
Hey, I'm trying to verify a JWT using a SPKI public key in a Next.js middleware like this:

```
const publicKey = await jose.importSPKI(process.env.PUBLIC_KEY!, "RS256");

const { payload, protectedHeader } = await jose.jwtVerify(authToken, publicKey);
```

The importSPKI function seems to rely on node:crypto, which can't be run in Vercel edge functions. Is there a way around this?

**Comment by panva** [comment]
You should reach out to in the nextjs project. This module has well defined runtime export targets that have been confirmed working years ago.
